assets/src/edit-story/components/panels/design/pageAttachment/pageAttachment.js

```diff
@@ -89,25 +121,43 @@ function PageAttachmentPanel() {
     [updateCurrentPageProperties, pageAttachment]
   );
 
-  const debouncedCTAUpdate = useDebouncedCallback((value) => {
-    updatePageAttachment({ ctaText: value });
+  const debouncedUpdate = useDebouncedCallback((props) => {
+    updatePageAttachment(props);
   }, 300);
 
   const [isInvalidUrl, setIsInvalidUrl] = useState(
     !isValidUrl(withProtocol(url || '').trim())
   );
 
   const isDefault = _ctaText === defaultCTA;
-  const showTextInput = Boolean(url) && !isInvalidUrl;
+  const hasValidUrl = Boolean(url) && !isInvalidUrl;
 
-  const handleChange = useCallback(
+  const handleChangeUrl = useCallback(
+    (value) => {
+      _setUrl(value);
+      debouncedUpdate({ url: value });
```

Comment:
Should we re-use the logic to default the icon as derived from the URL source as we do for links or is that not relevant for page attachments?

Reply:
Not sure, that was a question I had under the PR description, too :)

It has a decent default icon, so it might be OK here though unless there's a preference for trying to pull it as in case of Links. WDYT?

Reply:
I'm indifferent. @swissspidy?

Reply:
Hmm good question.

Let's leave the default icon for now and then raise with product during testing.